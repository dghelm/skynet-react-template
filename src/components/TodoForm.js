import { Input, Button } from "semantic-ui-react";
import { useState } from "react";
import { useStoreActions } from "easy-peasy";

function TodoForm() {
  const addTodo = useStoreActions((actions) => actions.todos.addTodo);
  const [value, setValue] = useState("");
  return (
    <>
      <Input
        placeholder="Todo item..."
        onChange={(e) => setValue(e.target.value)}
        value={value}
      />
      <Button
        onClick={() => addTodo({ text: value, done: false })}
        style={{ marginLeft: "10px" }}
      >
        Add Todo
      </Button>
    </>
  );
}

export default TodoForm;
