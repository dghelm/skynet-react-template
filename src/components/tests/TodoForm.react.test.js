import renderer from "react-test-renderer";
import TodoForm from "../TodoForm";

test("Check if forms renders correctly", () => {
  const component = renderer.create(TodoForm);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
