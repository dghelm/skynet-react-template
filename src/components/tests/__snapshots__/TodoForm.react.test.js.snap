// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if forms renders correctly 1`] = `null`;
